mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h

```diff
@@ -117,10 +126,18 @@ class GreedyRewriteConfig {
 ///
 /// Note: This method does not apply patterns to the region's parent operation.
 LogicalResult
+applyPatternsGreedily(Region &region, const FrozenRewritePatternSet &patterns,
+                      GreedyRewriteConfig config = GreedyRewriteConfig(),
+                      bool *changed = nullptr);
+/// Same as `applyPatternsAndGreedily` above with folding.
+inline LogicalResult
 applyPatternsAndFoldGreedily(Region &region,
                              const FrozenRewritePatternSet &patterns,
                              GreedyRewriteConfig config = GreedyRewriteConfig(),
-                             bool *changed = nullptr);
+                             bool *changed = nullptr) {
+  config.fold = true;
+  return applyPatternsGreedily(region, patterns, config, changed);
+}
 
```

Comment:
The API split makes sense, but then the GreedyRewriteConfig changes don't!

Reply:
The top one means it may fold, this one it must.

I could hide it further down and thread folding flag through along with config if that is preferred.

Reply:
> The top one means it may fold, this one it must.

The top one with the right config is equivalent to this one, why would we have a separate API here? Why specifically for fold and not for other parameters of the config?

Reply:
Mostly as it is already widely used and so makes possible to integrate without much fanfare. I can mark it as deprecated for removal post. But it seemed good way to avoid too much effort on downstream for relatively little code.

Reply:
Yes please mark as deprecated and plan for removal soon.

Reply:
Done. Should I migrate core users to the non-deprecated form too?

Reply:
I think we need to: otherwise we'll get warnings in tree.